src/components/serviceList.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Stack,
+  Tab,
+  Tabs,
+  ToggleButton,
+  Tooltip,
+  Typography,
+  useTheme,
+} from '@mui/material';
+import { useEffect, useMemo, useState } from 'react';
+import { Logo } from './logo';
+
+enum ServiceListMode {
+  All = 'all',
+  Popular = 'popular',
+}
+
+type ServiceType = {
+  name: string;
+  label: string;
+  logo: string;
+  tables: string[];
+};
+
+/**
+ * ServiceTypes type is the required shape of the services for the ServiceList
+ *
+ * @public
+ */
+export type ServiceTypes = Record<string, ServiceType>;
+
+interface Props {
+  services: ServiceTypes;
+  topServices: string[];
+  value?: string[];
+  onChange?: (value: string[]) => void;
+  fallbackLogoSrc?: string;
+}
+
+/**
+ * ServiceList component is multi-select form component for selecting services
+ * with an expandable view of all available services.
+ *
+ * @public
+ */
+export function ServiceList({
+  services,
+  topServices,
+  fallbackLogoSrc,
+  value = [],
+  onChange,
+}: Props) {
+  const { palette } = useTheme();
+
+  const [showServices, setShowServices] = useState<ServiceListMode.All | ServiceListMode.Popular>(
+    ServiceListMode.Popular,
+  );
+
+  // prefetch logos
+  useEffect(() => {
+    if (services) {
+      Object.values(services).forEach((service) => {
+        const img = new Image();
+        img.src = service.logo;
+      });
+    }
```

Comment:
Nice! But doe we need to have this `if`? Because according to the types it should always be defined

Reply:
great point! I think I had a second param in the dependency array earlier and there was a chance of an undefined services. but no longer 👍